pkg/api/v1/token_test.go

```diff
@@ -0,0 +1,155 @@
+package apiv1
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/beam-cloud/beta9/pkg/auth"
+	"github.com/beam-cloud/beta9/pkg/common"
+	"github.com/beam-cloud/beta9/pkg/repository"
+	"github.com/beam-cloud/beta9/pkg/types"
+	"github.com/labstack/echo/v4"
+	"github.com/tj/assert"
+)
+
+type MockTokenTestDetails struct {
+	backendRepo *repository.BackendRepository
+	server      *echo.Echo
+	httpDetails *repository.MockHttpDetails
+}
+
+func setupTokenEchoGroup() MockTokenTestDetails {
+	configManager, err := common.NewConfigManager[types.AppConfig]()
+	if err != nil {
+		log.Fatalf("Failed to create config manager: %v", err)
+	}
+
+	config := configManager.GetConfig()
+	md := repository.MockBackendWithValidToken()
+
+	e := echo.New()
+	e.Use(auth.AuthMiddleware(md.BackendRepo))
+	NewTokenGroup(e.Group("/token"), md.BackendRepo, config)
+
+	return MockTokenTestDetails{
+		backendRepo: &md.BackendRepo,
+		server:      e,
+		httpDetails: &md,
+	}
+}
+
+func TestListWorkspaceTokens(t *testing.T) {
+	testDetails := setupTokenEchoGroup()
+
+	tests := []struct {
+		name           string
+		prepares       func()
+		expectedStatus int
+	}{
+		{
+			name:           "Test list",
+			expectedStatus: 200,
+			prepares: func() {
+				// First query is for token select on the auth middleware
+				testDetails.httpDetails.AddExpectedDBTokenSelect(testDetails.httpDetails.Mock, *testDetails.httpDetails.TokenForTest.Workspace, testDetails.httpDetails.TokenForTest)
+				// Second query is for the workspace check inside the token list method
+				addWorkspaceRowSelectQuery(testDetails.httpDetails)
+				// Third query is for the token list backend
+				testDetails.httpDetails.AddExpectedDBTokenSelect(testDetails.httpDetails.Mock, *testDetails.httpDetails.TokenForTest.Workspace, testDetails.httpDetails.TokenForTest)
+			},
+		},
+		{
+			name: "Test error occurs in db",
+			prepares: func() {
+				testDetails.httpDetails.AddExpectedDBTokenSelect(testDetails.httpDetails.Mock, *testDetails.httpDetails.TokenForTest.Workspace, testDetails.httpDetails.TokenForTest)
```

Comment:
Same comment as above, I don't really follow the structure of these tests - what exactly are we testing here?